homeassistant/components/switch/ecoal_boiler.py

```diff
@@ -0,0 +1,104 @@
+"""
+Allows to configuration ecoal (esterownik.pl) pumps as switches.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/switch.ecoal_boiler/
+"""
+import logging
+from typing import Optional
+
+import voluptuous as vol
+
+from homeassistant.components.switch import PLATFORM_SCHEMA, SwitchDevice
+from homeassistant.components.ecoal_boiler import DATA_ECOAL_BOILER
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['ecoal_boiler']
+
+# Available ids
+PUMP_IDNAMES = (
+    "central_heating_pump",
+    "domestic_hot_water_pump",
+    "central_heating_pump2",
+)
+
+ENABLED_SCHEMA = {}
+for pump_id in PUMP_IDNAMES:
+    ENABLED_SCHEMA[vol.Optional(pump_id)] = cv.string
+CONF_ENABLE = "enable"
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {vol.Optional(CONF_ENABLE): ENABLED_SCHEMA}
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up switches based on ecoal interface."""
+    ecoal_contr = hass.data[DATA_ECOAL_BOILER]
+    config_enable = config.get(CONF_ENABLE, {})
+    # _LOGGER.debug("config_enable: %r", config_enable)
+    switches = []
+    for pump_id in PUMP_IDNAMES:     # pylint: disable=W0621
+        name = config_enable.get(pump_id)
+        if name:
+            switches.append(EcoalSwitch(ecoal_contr, name, pump_id))
+    add_entities(switches, True)
+
+
+class EcoalSwitch(SwitchDevice):
+    """Representation of Ecoal switch."""
+
+    def __init__(self, ecoal_contr, name, state_attr):
+        """
+        Initialize switch.
+
+        Sets HA switch to state as read from controller.
+        """
+        self._ecoal_contr = ecoal_contr
+        self._name = name
+        self._state_attr = state_attr
+        # NOTE: Ecoalcotroller holds convention that same postfix is used
```

Comment:
Ecoalcontroller

Reply:
We can remove the `NOTE:` annotation, and just use regular comment `#`.

Reply:
Done